docs/001_develop/03_client-capabilities/010_filters/foundation-filters.mdx

```diff
@@ -0,0 +1,437 @@
+            Start Period
+          </rapid-text-field>
+          <rapid-text-field type="datetime-local" :value=${sync((x) => x.endDate)}>
+            End Period
+          </rapid-text-field>
+          <rapid-button @click=${(x) => x.checkWindow()}>Check</rapid-button>
+          <span>${(x) => x.resultString}</span>
+        </div>
+      `,
+    })
+    export class MyElement extends GenesisElement {
+      @observable startDate: string;
+      @observable endDate: string;
+      @observable resultString: string;
+      @TimeWindow timeWindow: TimeWindow;
+      checkWindow() {
+        this.resultString = this.timeWindow.filter({start: this.startDate, end: this.endDate})
+          ? 'Exists within the start and end period'
+          : 'Doesnot fall between start and end period';
+      }
+    }
+
+    ```
+
+The function version is pretty self-explanatory, and you may be wondering why you would use the DI injected version. The
+reason is at the point of calling a filter, either directly or indirectly as (say) part of a chain, you may not know what
+data or state from the application it needs to provide its outcome. The DI versions encapsulate this complexity, which aids later refactoring, and leaves callers free to pass only primitive input parameters where possible.
+
+### DI use cases
+
+Integrating filters using Dependency Injection (DI) enables dynamic configurations and simplifies reusability across components. Below is an example of using userTargetingFilter with DI for precise user-specific logic.
+
+```ts
+import { UserTargeting } from '@genesislcap/foundation-filters';
+...
+@UserTargeting userTargeting: UserTargeting;
+...
+const outcome = this.userTargeting.filter({
+  profiles: ['ADMIN']
+});
+```
+
+Here we've used the `UserTargeting` filter without the need to provide the current user or other data points to the underlying function directly. If we source the current user from changes over time, the author of the filter can refactor centrally.
+
+All DI-based filters implement the ClientFilter<!-- TODO:(./docs/api/foundation-filters.clientfilter.md)--> interface; however, they may provide other APIs for convenience to abstract underlying implementation details. For example, `UserTargeting` provides a `hasAdminProfile` API, so the previous example can be re-written as:
+
+```ts
+import { UserTargeting } from '@genesislcap/foundation-filters';
+...
+@UserTargeting userTargeting: UserTargeting;
+...
+const outcome = this.userTargeting.hasAdminProfile();
+```
+
+This is quite a simplistic example, but we hope it helps to highlight that common parameters can be predefined in the
+filters themselves where needed. It is also worth mentioning that DI-based filters are registered as transient, meaning
+you will get a new instance for every dependency request or container.get(). The ClientFilterRunner used to run chained
+filters is also transient.
+
+## Combining filters with the runner method
```

Comment:
Move this section to the bottom after the examples, I think it makes sense to introduce the filters first

Reply:
done in lstest commit